src/GovUk.Education.ExploreEducationStatistics.Admin/Services/MetaGuidanceService.cs

```diff
@@ -0,0 +1,186 @@
+                releaseSubjects
+                    .OrderBy(subject => subject.Subject.Name)
+                    .Select(async releaseSubject =>
+                    {
+                        var subject = releaseSubject.Subject;
+                        var geographicLevels = await GetGeographicLevels(subject.Id);
+                        var timePeriods = await GetTimePeriods(subject.Id);
+                        var variables = GetVariables(subject.Id);
+                        return new MetaGuidanceSubjectViewModel
+                        {
+                            Id = subject.Id,
+                            Content = releaseSubject.MetaGuidance,
+                            Filename = dataFiles[subject.Id].Filename,
+                            Name = subject.Name,
+                            GeographicLevels = geographicLevels,
+                            TimePeriods = timePeriods,
+                            Variables = variables
+                        };
+                    })
+            )).ToList();
+        }
+
+        private async Task<MetaGuidanceSubjectTimePeriodsViewModel> GetTimePeriods(Guid subjectId)
+        {
+            var orderedObservations = _statisticsDbContext
+                .Observation
+                .Where(observation => observation.SubjectId == subjectId)
+                .OrderBy(observation => observation.Year)
+                .ThenBy(observation => observation.TimeIdentifier);
+
+            var first = await orderedObservations.FirstOrDefaultAsync();
+            var last = await orderedObservations.LastOrDefaultAsync();
+            return new MetaGuidanceSubjectTimePeriodsViewModel(first?.GetTimePeriod(), last?.GetTimePeriod());
+        }
+
+        private async Task<List<GeographicLevel>> GetGeographicLevels(Guid subjectId)
+        {
+            return await _statisticsDbContext
+                .Observation
+                .Where(observation => observation.SubjectId == subjectId)
+                .Select(observation => observation.GeographicLevel)
+                .Distinct()
+                .ToListAsync();
+        }
+
+        private List<LabelValue> GetVariables(Guid subjectId)
+        {
+            var filters = _filterService.FindMany(filter => filter.SubjectId == subjectId)
+                .Select(filter =>
+                    new LabelValue(
+                        string.IsNullOrWhiteSpace(filter.Hint) ? filter.Label : $"{filter.Label} - {filter.Hint}",
+                        filter.Name))
+                .ToList();
+
+            var indicators = _indicatorService.GetIndicators(subjectId)
+                .Select(indicator => new LabelValue(indicator.Label, indicator.Name));
+
+            return filters.Concat(indicators)
+                .OrderBy(labelValue => labelValue.Value)
+                .ToList();
+        }
+
+        private static MetaGuidanceViewModel BuildViewModel(Release release,
+            List<MetaGuidanceSubjectViewModel> subjects)
```

Comment:
Rather than passing subjects in, wouldn't it be easier to just call `GetSubjects` from within here?

Reply:
Cool, done it. Not sure why it was like this! 👍